src/tools/compiletest/src/runtest.rs

```diff
@@ -1027,12 +1027,31 @@ impl<'test> TestCx<'test> {
     }
 
     fn set_revision_flags(&self, cmd: &mut Command) {
+        // Normalize revisions to be lowercase and replace `-`s with `_`s.
+        // Otherwise the `--cfg` flag is not valid.
+        let normalize_revision = |revision: &str| revision.to_lowercase().replace("-", "_");
```

Comment:
Wait, what happens if a test writer specifies both `foo-bar` and `foo_bar` (an edge case that's unlikely to be hit, but the less jank the better)? Do we try to emit `--cfg foo_bar` twice?

Reply:
Pre-existing, but yes I think so.

Reply:
The compiler does handle duplicate `--cfg` arguments fine; but that would never be the case here, only one `--cfg` argument is set per revision. The only issue with similar names is that inside the code it would be impossible to distinguish between the two revisions.